packages/create-cloudflare/src/common.ts

```diff
@@ -315,3 +315,21 @@ export async function initializeGit(cwd: string) {
 		await runCommand(`git init`, { useSpinner: false, silent: true, cwd });
 	}
 }
+
+export async function getProductionBranch(cwd: string) {
+	try {
+		const producitonBranch = await runCommand(
+			"git rev-parse --abbrev-ref HEAD",
+			{
+				silent: true,
+				cwd,
+				useSpinner: false,
+				captureOutput: true,
+			}
+		);
+
+		return producitonBranch.trim();
+	} catch (err) {}
+
+	return "main";
```

Comment:
What happens if the user chose to not use git? Will the default `main` still work or should we throw a better error than the potential crash would?

Reply:
Pages needs a "production branch" even if git isn't used (e.g. for direct uploads). It is used as a tag for the production environment. So I think this is OK.
We should move this return statement inside the catch block, though, rather than having an empty catch block.

Reply:
It'll default to the current behavior. We need to supply a `--productionBranch` argument to the pages create command, otherwise the call to `wrangler pages project create` will fail when it demands interactive input.